supporting_scripts/course-scripts/quick-course-setup/utils.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import configparser
+import logging
+from typing import Dict, Any, List
+
+config = configparser.ConfigParser()
+config.read('config.ini')
+
+server_url: str = config.get('Settings', 'server_url')
+admin_user: str = config.get('Settings', 'admin_user')
+admin_password: str = config.get('Settings', 'admin_password')
+
+def login_as_admin(session: requests.Session) -> None:
+    """Authenticate as an admin using the provided session."""
+    authenticate_user(admin_user, admin_password, session)
+
+def add_user_to_course(session: requests.Session, course_id: int, user_group: str, user_name: str) -> None:
+    """Add a user to a specified course and group."""
+    url: str = f"{server_url}/api/courses/{course_id}/{user_group}/{user_name}"
+    response: requests.Response = session.post(url)
+    if response.status_code == 200:
+        logging.info(f"Added user {user_name} to group {user_group}")
+    else:
+        logging.error(f"Could not add user {user_name} to group {user_group}")
+
+def authenticate_user(username: str, password: str, session: requests.Session = requests.Session()) -> requests.Response:
+    """Authenticate a user and return the session response."""
+    url: str = f"{server_url}/api/public/authenticate"
```

Comment:
just a small idea: you could add `/api` also to the `server_url`

Reply:
did it for client aswell